src/protocol.rs

```diff
@@ -1,23 +1,130 @@
+#![cfg(feature = "rpc")]
 //! [`ProtocolHandler`] implementation for the docs [`Engine`].
 
+use std::{path::PathBuf, sync::Arc};
+
 use anyhow::Result;
 use futures_lite::future::Boxed as BoxedFuture;
 use iroh::{endpoint::Connecting, protocol::ProtocolHandler};
+use iroh_blobs::net_protocol::{Blobs, ProtectCb};
+use iroh_gossip::net::Gossip;
+use quic_rpc::server::{ChannelTypes, RpcChannel};
 
-use crate::engine::Engine;
+use crate::{
+    engine::{DefaultAuthorStorage, Engine},
+    rpc::proto::{Request, RpcService},
+    store::Store,
+};
 
-impl<D: iroh_blobs::store::Store> ProtocolHandler for Engine<D> {
+impl<S: iroh_blobs::store::Store> ProtocolHandler for Docs<S> {
     fn accept(&self, conn: Connecting) -> BoxedFuture<Result<()>> {
-        let this = self.clone();
+        let this = self.engine.clone();
         Box::pin(async move { this.handle_connection(conn).await })
     }
 
     fn shutdown(&self) -> BoxedFuture<()> {
-        let this = self.clone();
+        let this = self.engine.clone();
         Box::pin(async move {
             if let Err(err) = this.shutdown().await {
                 tracing::warn!("shutdown error: {:?}", err);
             }
         })
     }
 }
+
+/// Docs protocol.
+#[derive(Debug, Clone)]
+pub struct Docs<S> {
+    engine: Arc<Engine<S>>,
```

Comment:
`Engine` is already clone, that's a lot of nested `Arc`s now. Maybe remove some `Arc`s in `Engine` and remove the `Arc`s in there?

Reply:
It's just indiretion, not dynamic dispatch. I was going to look into making Engine !Clone next, but I don't think it's the end of the world.

Reply:
OK, made Engine !Clone.